rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -167,23 +170,26 @@ def _recursively_initialize_decoders(
                     address_results.update(self.decoders[class_name].addresses_to_decoders())
                     rules_results.extend(self.decoders[class_name].decoding_rules())
                     enricher_results.extend(self.decoders[class_name].enricher_rules())
+                    post_decoding_results.extend(self.decoders[class_name].post_decoding_rules())  # noqa: E501
                     self.all_counterparties.update(self.decoders[class_name].counterparties())
 
-                recursive_addrs, recursive_rules, recurisve_enricher_results = self._recursively_initialize_decoders(full_name)  # noqa: E501
+                recursive_addrs, recursive_rules, recurisve_enricher_results, recursive_post_decoding_results = self._recursively_initialize_decoders(full_name)  # noqa: E501
                 address_results.update(recursive_addrs)
                 rules_results.extend(recursive_rules)
                 enricher_results.extend(recurisve_enricher_results)
+                post_decoding_results.extend(recursive_post_decoding_results)
 
-        return address_results, rules_results, enricher_results
+        return address_results, rules_results, enricher_results, post_decoding_results
 
     def initialize_all_decoders(self) -> None:
         """Recursively check all submodules to get all decoder address mappings and rules
         """
         self.decoders: dict[str, 'DecoderInterface'] = {}
-        address_result, rules_result, enrichers_result = self._recursively_initialize_decoders(self.chain_modules_root)  # noqa: E501
+        address_result, rules_result, enrichers_result, post_decoding_results = self._recursively_initialize_decoders(self.chain_modules_root)  # noqa: E501
         self.address_mappings = address_result
         self.event_rules.extend(rules_result)
         self.token_enricher_rules.extend(enrichers_result)
+        self.post_decoding_rules.extend(post_decoding_results)
```

Comment:
We will also need to somehow specify order of processing here, as we need to guarantee that a certain decoder's post processing happens before some other one's.

For example I can see that you may need to require uniswap to run before cowswap or 1inch. I gave an idea for this below in the `post_decoding_rules` function